packages/block-editor/src/components/block-settings-menu/index.js

```diff
@@ -23,113 +26,89 @@ import BlockUnknownConvertButton from './block-unknown-convert-button';
 import __experimentalBlockSettingsMenuFirstItem from './block-settings-menu-first-item';
 import __experimentalBlockSettingsMenuPluginsExtension from './block-settings-menu-plugins-extension';
 
-export function BlockSettingsMenu( { clientIds, onSelect } ) {
+export function BlockSettingsMenu( { clientIds } ) {
 	const blockClientIds = castArray( clientIds );
 	const count = blockClientIds.length;
 	const firstBlockClientId = blockClientIds[ 0 ];
 
 	return (
 		<BlockActions clientIds={ clientIds }>
 			{ ( { onDuplicate, onRemove, onInsertAfter, onInsertBefore, canDuplicate, isLocked } ) => (
-				<Dropdown
-					contentClassName="editor-block-settings-menu__popover block-editor-block-settings-menu__popover"
-					position="bottom right"
-					renderToggle={ ( { onToggle, isOpen } ) => {
-						const toggleClassname = classnames( 'editor-block-settings-menu__toggle block-editor-block-settings-menu__toggle', {
-							'is-opened': isOpen,
-						} );
-						const label = isOpen ? __( 'Hide options' ) : __( 'More options' );
```

Comment:
Note: We previously would change the label of the block settings toggle button to reflect whether the menu was opened or closed. This is not supported in DropdownMenu. It's unclear whether this would add value anyways. I would appreciate accessibility feedback here, noting also that the toggle button does apply specific props to indicate the expanded status of the menu (`aria-expanded` as it did before, and an additional `aria-haspopup` which did not exist for BlockSettingsMenu, but is applied in DropdownMenu).

If it should be supported, then support should be made available via `DropdownMenu` to not limit the `label` prop to a single static string value, but rather as a pair, or a function which returns a string given the open state.

Reply:
I refactored it to allow using different values depending on `isOpen` value to make it work the same as before. However, we can revert that change as soon as we get accessibility feedback to use the revised approach.

Reply:
If I get it correctly, this refers to the current button label (and tooltip) changing from “Show more tools…” to “Hide more tools…” depending on the state, correct? Personally, I’m wary of dynamically changing the accessible name of a control. A name is a name, it shouldn’t change. The actual _state_ should be represented by other means. So if I’m getting it correctly, it would be ok for me.

Reply:
Updated as discussed in 07d3716750f094fb72fc29b4dfa6303d3ebc4578.